### ISSUE TITLE
HELP TRANSLATE THE APP

### ISSUE THREAD

**Issue by 7LPdWcaW** [7LPdWcaW/GrowTracker-Android#116]
If you enjoy the app and want to give back, why not help translate it into your native language?

Sign up as a translator here to help https://www.transifex.com/growutils/growtracker/dashboard/

**Comment by 7LPdWcaW** [comment]
![ES - Spanish](https://img.shields.io/endpoint.svg?url=https://transifex.quelltext.eu/badge/growutils/project/growtracker/language/es/translated.json?)

![DE - German](https://img.shields.io/endpoint.svg?url=https://transifex.quelltext.eu/badge/growutils/project/growtracker/language/de_DE/translated.json?)

**Comment by EmmanuelMess** [comment]
"Notes" is a repeated a few times, is that an error?

**Comment by 7LPdWcaW** [comment]
Strings have not yet been optimised so there may be some duplicates, also context may change, but hopefully not. I'll do some optimisations and update the source strings

**Comment by EmmanuelMess** [comment]
@7LPdWcaW 
"Default garden to show on open, currently"
is missing a `%s`, almost made an incorrect translation, please correct.

**Comment by EmmanuelMess** [comment]
@7LPdWcaW 
You seem to mix "MB", "mb" and "Mb", those mean different things.

**Comment by 7LPdWcaW** [comment]
> @7LPdWcaW
> You seem to mix "MB", "mb" and "Mb", those mean different things (except "mb", I don't think it means anything).

These have been fixed

**Comment by Maxtille** [comment]
Hello there :v:, 
I'd like to help on translation, so I've made a language request on transifex. :fr: 
If you accept it I will ask to join the team :wink:

**Comment by EmmanuelMess** [comment]
@7LPdWcaW Are you able to integrate GitHub and Transifex as described [here](https://docs.transifex.com/transifex-github-integrations/github-tx-ui)?

**Comment by Maxtille** [comment]
@7LPdWcaW  100% translated or reviewed ? Cause I think it would be an idea to use reviewing while able to check strings directly within the application

**Comment by 7LPdWcaW** [comment]
@Maxtille I went for translated so that I can do merges before they are reviewed for you guys to test (also i will have to rely on you all to tell me when it all looks good and i will set reviewed to 100% as I can only speak english ;))

**Comment by Maxtille** [comment]
Hey there, I just finished french translation yesterday but it doesn't seem to have automatically sent you a PR, or I can't see it :wink:

**Comment by Maxtille** [comment]
Hey @7LPdWcaW :smile: 
I just modified french strings that were inadequate, I was not able to review because in transifex you cannot review string you translated ... which is not stupid at all, unless you work alone on a language :rofl: 
But never mind, modifying is very sufficient for me. :wink:

**Comment by 7LPdWcaW** [comment]
@Maxtille i've made you a reviewer for French

**Comment by 9YbQiuEohUu1** [comment]
I would like to help and have already sent a request.

**Comment by 7LPdWcaW** [comment]
@9YbQiuEohUu1 i've invited you

**Comment by EmmanuelMess** [comment]
What does "cure a plant" mean?

**Comment by 9YbQiuEohUu1** [comment]
> 
> 
> What does "cure a plant" mean?

It's a preservation technique, something between drying and fermentation.

**Comment by 7LPdWcaW** [comment]
@EmmanuelMess curing is to let 'mature' like leaving whisky in the barrel for many years

**Comment by 9YbQiuEohUu1** [comment]
A couple of clarifications:
Should PPM be spelled ppm?
Similarly, is MiB actually MB?
"Are you sure you want to delete  images, are you sure?"

Next, what is cutting? I have not seen this stage in the literature anywhere and do not know how to translate it correctly.

Finally, what is "fail safe"?

Thanks.

**Comment by 9YbQiuEohUu1** [comment]
BTW, finished my part.

**Comment by 7LPdWcaW** [comment]
@9YbQiuEohUu1

>Should PPM be spelled ppm?

No, PPM stands for "parts per million"

> Similarly, is MiB actually MB?

No, we're using Mebibytes (powers of 2 1024 kibibytes = 1 mebibyte)

>"Are you sure you want to delete images, are you sure?"

I will fix this

>Next, what is cutting? 

A cutting is when you take a small part of a mother plant and use as a clone to grow a new plant from. This usually replaces seedlings (e.g seedling -> vegetation and cutting -> vegetation)

>Finally, what is "fail safe"?

Fail safe is a way to safely fail an encryption authentication, for example, under duress you may give a fake password that reveals nothing rather than the actual data you're protecting (https://en.wikipedia.org/wiki/Fail-safe)

**Comment by 9YbQiuEohUu1** [comment]
> 
> No, PPM stands for "parts per million"
> 
That's what I figured, but the abbreviation for parts per million is spelled in lowercase all over the Internet.
> 
> No, we're using Mebibytes (powers of 2 1024 kibibytes = 1 mebibyte)
> 
This is literally the first time I've heard about such a thing O_o Thaks for clarifying.
> 
> A cutting is when you take a small part of a mother plant and use as a clone to grow a new plant from. This usually replaces seedlings (e.g seedling -> vegetation and cutting -> vegetation)
I'm afraid I didn't quite get it. Is it a stage in the life cycle of a mother plant or a clone? Either way, what's the duration of this stage? I actually have no experience of cloning, so to me it sounds as an action rather than a stage that lasts for a certain period.
> 
> Fail safe is a way to safely fail an encryption authentication, for example, under duress you may give a fake password that reveals nothing rather than the actual data you're protecting (https://en.wikipedia.org/wiki/Fail-safe)

This is what I thought. However, it seems that the "fail-safe" somewhat confusing as it describes a different concept. Maybe it would be more appropriate to call it plausible deniability or a decoy password?

**Comment by 7LPdWcaW** [comment]
>That's what I figured, but the abbreviation for parts per million is spelled in lowercase all over the Internet.

I try to be consistent with abbreviations with them being upper-cased

> I'm afraid I didn't quite get it. Is it a stage in the life cycle of a mother plant or a clone? Either way, what's the duration of this stage? I actually have no experience of cloning, so to me it sounds as an action rather than a stage that lasts for a certain period.

Yes, cutting would be treated as a stage of growth, in the same way a "seedling" is a stage of growth, except a seedling comes from a seed where as a cutting is taken from a more mature plant.

The duration is usually around 1-2 weeks less than a seedling as you're skipping the bit where it goes from germination to vegetation with cutting (rooting) into vegetation. heres more info https://www.growweedeasy.com/cloning-marijuana-guide

>This is what I thought. However, it seems that the "fail-safe" somewhat confusing as it describes a different concept. Maybe it would be more appropriate to call it plausible deniability or a decoy password?

"Decoy mode/password" works fine

**Comment by 9YbQiuEohUu1** [comment]
> 
> I try to be consistent with abbreviations with them being upper-cased
> 
What about l and ml, then? :)
> Yes, cutting would be treated as a stage of growth, in the same way a "seedling" is a stage of growth, except a seedling comes from a seed where as a cutting is taken from a more mature plant.
> 
> The duration is usually around 1-2 weeks less than a seedling as you're skipping the bit where it goes from germination to vegetation with cutting (rooting) into vegetation. heres more info https://www.growweedeasy.com/cloning-marijuana-guide
> 
So the stage is for a clone, not for a mother plant? e.g., planted->cutting ->vegetation instead of planted->germination->seedling->vegetation?

**Comment by 7LPdWcaW** [comment]
> What about l and ml, then? :)

Lowercase is acceptable for Metric https://en.wikipedia.org/wiki/Metric_system

>So the stage is for a clone, not for a mother plant? e.g., planted->cutting ->vegetation instead of planted->germination->seedling->vegetation?

Correct. I believe I have both, one as stage, and one as an action (i.e i have taken a cutting from this plant, and now it is a new plant that is growing from a cutting)

**Comment by 9YbQiuEohUu1** [comment]
> 
> Lowercase is acceptable for Metric https://en.wikipedia.org/wiki/Metric_system
> 
Can we make an exception for ppm? I know that parts-per notation is not a part of SI and everything, but it's so adorably consistent in all scientific papers and technical documentation... It will be like a heroic act of sacrificing your personal consistency in the name of the Universal Consistency (also known by its uppercase abbreviation of UC).
> Correct. I believe I have both, one as stage, and one as an action (i.e i have taken a cutting from this plant, and now it is a new plant that is growing from a cutting)
Understood, thank you. Still no clue about an adequate translation for it, so we'll have to make do with the literal one for now :(

**Comment by 7LPdWcaW** [comment]
> Can we make an exception for ppm 

Absolutely. I was playing with the idea of re-writing the EC/ppm functionality to include more options (like with the delivery/measurement units) so will keep this in mind when I do that

**Comment by 9YbQiuEohUu1** [comment]
Tried the alpha, noticed that there are no strings for card style description ("Select card style, currently X") and for styles themselves. 

Also noticed that stage names are duplicated (?) in strings 107-113 & 180 vs 120-125 & 179. Hope it helps.

**Comment by 7LPdWcaW** [comment]
@9YbQiuEohUu1 The issue with the labels is that the ":" is needed and is being referenced in xml code which does not allow me to concat strings, and I dont want to set it in code :P

**Comment by 9YbQiuEohUu1** [comment]
Found more gaps in translation:
![02](https://user-images.githubusercontent.com/50482952/66032336-eae93b80-e50d-11e9-9da5-d212c189a24d.PNG)
![01](https://user-images.githubusercontent.com/50482952/66032337-eae93b80-e50d-11e9-8b38-79c6f9179ac7.PNG)
Also, the card names are still untranslated (origial, compact, extreme)

**Comment by 7LPdWcaW** [comment]
@9YbQiuEohUu1 this should now be fixed, transifex didnt pick up some translations I added

**Comment by 9YbQiuEohUu1** [comment]
Looks like I now can review translations but cannot translate anymore (the button is greyed out)

**Comment by 7LPdWcaW** [comment]
@9YbQiuEohUu1 grr thats because transifex only allows reviewers to edit translations. i will change everyone to a reviewer instead

**Comment by 7LPdWcaW** [comment]
@9YbQiuEohUu1 could you post a screenshot? you should be able to translate as you're a reviewer for both UK and RU

**Comment by 9YbQiuEohUu1** [comment]
![Screenshot_2019-10-03 GrowTracker Transifex Editor](https://user-images.githubusercontent.com/50482952/66108252-dc148e80-e5ca-11e9-8e6c-0a612a5bf1e1.png)

**Comment by 7LPdWcaW** [comment]
@9YbQiuEohUu1 ahh ok i think you have to click the white buttons "1" and "other" before it will let you save as they are "plurals" (strings for 1 of something, or multiple like "day" and "days")

try that

**Comment by 9YbQiuEohUu1** [comment]
I wasn't aware of this,, thanks. 
Another question then. I see there are also "few" and "many" buttons. Is it posible to add more than one type of plurals? Russian and Ukrainian have separate forms for 2-4 and 5-9.

**Comment by 7LPdWcaW** [comment]
yes, not sure if the english is required or if you can just add them

**Comment by 9YbQiuEohUu1** [comment]
I have added the translations as follows:
1 - for 1
few - for 2-4
many - for 5-9
other - for 5-9

However, I've noticed that the English strings are as follows:
1 - shows singular form
few - shows singular form
many - shows singular form
other - shows plural form

So I'm not sure how it will work. Will check out the next build.

**Comment by 7LPdWcaW** [comment]
👌 in english for time units we only really have 2 versions, "day" and "days" as "days" just means more than 1

**Comment by 9YbQiuEohUu1** [comment]
That I know) My point was that both "few" and "many" should correspond to "days," not "day," but I have no idea how this works in Transifex, so I'll have to wait for the next build.

**Comment by 9YbQiuEohUu1** [comment]
Just tested the latest alpha and all numerals work fine now. Some more issues though:
1. Strings for card types are still missing.
2. The setting looks like this now:
![Screenshot_20191003-211433_1](https://user-images.githubusercontent.com/50482952/66153224-c8494680-e623-11e9-9cd0-52e92f28ace5.png)

3. Confirmation dialog and quit button isn't translated
![Screenshot_20191003-210316_1](https://user-images.githubusercontent.com/50482952/66153300-f7f84e80-e623-11e9-893a-b275d83d80be.png)

**Comment by 9YbQiuEohUu1** [comment]
Checked the latest alpha, found a couple of minor issues:
1. No string for "No chart data available" 

2. Strings 68, 69, and 70 lack the options for different types of plurals (like in string 294). Can they be added there?

Also, just out of curiosity, where is the "Cured for x days" string used in the app? Couldn't find it.

**Comment by 7LPdWcaW** [comment]
@9YbQiuEohUu1 
>Cured for x days 

is only used when a plant has been set to 'harvested' and the plant has had a stage change of `CURING` 

>%s days 

can be deleted in favour of the new plurals i think

>Days %s

Is used when displaying the current stage, i.e "5 days curing". Im not sure if this should be `%s days %s` for placeholders (plus plurals) or if that can even be translated nicely

**Comment by EmmanuelMess** [comment]
What is "ScrOG Tuck"?

**Comment by 7LPdWcaW** [comment]
@EmmanuelMess "tuck" as in tucking something under something else, and "ScrOG" as in "screen of green" (more info here https://www.growweedeasy.com/scrog-tutorial) Id say you can just translate the word "tuck" as "ScrOG" is a term in its own right and probably isnt translatable

**Comment by EmmanuelMess** [comment]
The string "w" is "weeks" but I don't know how to make "semanas" into a single letter, any ideas? Can you say how long the string can be?

**Comment by 7LPdWcaW** [comment]
@EmmanuelMess String can be any length, but as short as possible.

**Comment by EmmanuelMess** [comment]
> @EmmanuelMess String can be any length, but as short as possible.

And what about pluralization? A single letter usually isn't pluralized, but longer abbreviations need to pluralized.

**Comment by 7LPdWcaW** [comment]
@EmmanuelMess Im not sure. In English, we typically use `w` for `week/s`, or `wk` or `wks`, but the plural is implied because of the number (i.e 1w = 1 week, 2w = 2 weeks)

**Comment by EmmanuelMess** [comment]
> @EmmanuelMess Im not sure. In English, we typically use w for week/s, or wk or wks, but the plural is implied because of the number (i.e 1w = 1 week, 2w = 2 weeks)

I translated as singular, so "2y" translates to "2 año" which is wrong, it should be "2 años".

**Comment by EmmanuelMess** [comment]
Last question: What is an "image disk"?

In "Image disk and memory cache cleared".

**Comment by EmmanuelMess** [comment]
@9YbQiuEohUu1 
> This is what I thought. However, it seems that the "fail-safe" somewhat confusing as it describes a different concept. Maybe it would be more appropriate to call it plausible deniability or a decoy password?

I translated it as "encrypted false compartment"

**Comment by 7LPdWcaW** [comment]
>Last question: What is an "image disk" and "memory cache"?

>In "Image disk and memory cache cleared".

it's more image "disk and memory"

like disk cache and memory cache for images

**Comment by 9YbQiuEohUu1** [comment]
Another minor localization issue: the word "ago" is not translated in the last action preview of the plant tab 

![ff](https://user-images.githubusercontent.com/50482952/68493908-0dc9e680-0256-11ea-934e-6ae2a1205598.PNG)

**Comment by 9YbQiuEohUu1** [comment]
Minor error: "File successfully imported from file"
Also,

> the word "ago" is not translated in the last action preview of the plant tab

is still there

**Comment by 9YbQiuEohUu1** [comment]
> Minor error: "File successfully imported from file"

Now it's "**Plantsp** successfully imported from file"

**Comment by Maxtille** [comment]
I don't really get the meanings of 

- Budding (diiferent from flowering ?)

- Ripening

- Supercrop 

- and Monstercrop ...

 maybe you could explain a little bit so that we can translate it for the best ?

**Comment by inverseTrig** [comment]
I believe "budding" refers to flowering. "Ripening" usually refers to when a fruit is becoming ripe (or ready to be harvested). In the case of our plant, I believe it would be easier to understand not as a fruit, but as a flower. "Supercrop" is to intentionally harm, usually the branch or stem, of the plant to induce stress. Some people pinch the branches until they hear it pop or break. I am unsure about "monstercrop".

**Comment by 7LPdWcaW** [comment]
> I don't really get the meanings of
> 
> * Budding (diiferent from flowering ?)
> * Ripening
> * Supercrop
> * and Monstercrop ...
> 
> maybe you could explain a little bit so that we can translate it for the best ?

For the additional (optional) stages, I used this as a guide - https://www.saferbrand.com/articles/plant-growth-stages

Super cropping is when you damage the plant to induce stress as @inverseTrig said

Monster cropping is when a clone is taken from a plant currently in flower

**Comment by 9YbQiuEohUu1** [comment]
Where string 239 "Average" is used? Can you give an example of how it is supposed to look in the app?

**Comment by 7LPdWcaW** [comment]
#239 "Average %s" is used for TDS - So what ever is selected or has data (EC, PPM500, PPM700) will say "Average EC" etc

**Comment by 9YbQiuEohUu1** [comment]
Is it possible to create a pull request manually so that the latest edition of translation could make it into the nearest alpha?

**Comment by 7LPdWcaW** [comment]
~~Seems like Transifex thinks ru/uk are already up-to-date, are you talking about another language?~~ nevermind, transifex was just slow at creating the PRs!

**Comment by 9YbQiuEohUu1** [comment]
Actually, it looks like the bot creates a PR once the translation reaches 100%. So if I edit an already completed translation it does nothing (because the progress stayed at 100%).

**Comment by aha999** [comment]
I can recommend [Weblate](https://weblate.org/), as it's popular, the platform itself is open source (compared to others), can be self hostable, and for open source projects like yours, they host translations for free. Weblate automatically merges translations and if you want you can have a widget shown in github readme, that displays how much percentage each applied language is translated to in real time.  Currently you have to accept new translations via PR. With Weblate, translations can be merged automatically.

- [set up your libre project](https://hosted.weblate.org/hosting/) 

screenshot:

![screenshot](https://user-images.githubusercontent.com/50620416/119900111-57ffdf00-bf44-11eb-95e6-892f025a7627.png)

**Comment by danilo-jlle** [comment]
Hi.
I would like to translate it to Portuguese (Brazil).
I already asked to join the Team.
Any help would be appreciated.
Thanks.

My profile: https://www.transifex.com/user/profile/danilocsilva/

**Comment by aerowolf** [comment]
Hi.
I'd like to help on translation it to Chinese (China, zh_CN).
I've made a language request on transifex.
Any help would be appreciated.
Thanks.
